lib/winston/transports/file.js

```diff
@@ -143,9 +142,21 @@ module.exports = class File extends TransportStream {
       this._endStream(() => this._rotateFile());
     }
 
+    if (this._drain) {
```

Comment:
I am trying all my test cases and you seem to be right, this might be unnecessary. I haven't made a huge effort to investigate if the log function of the file transport might be called again if we've been told to wait for a drain event. This was to ensure we don't call .write on the stream again until its ready but if something else higher up prevents that making this check unnecessary than it can be removed.

Reply:
The `_writev` method of [`winstonjs/winston-transport`](https://github.com/winstonjs/winston-transport) seems to call the `log` method multiple times. I've managed to get into the `if`-statement by changing the `setInterval` part of the script I provided in my previous comment with the following:

```js
const infos = new Array(10).fill().map(() => ({
  level: 'info',
  //
  // Creates a message which is just under the `highWaterMark`limit.
  // See: https://nodejs.org/api/stream.html#stream_constructor_new_stream_writable_options
  // This should NOT work on `master`, but does work with @mempf's patch.
  //
  message: 'a'.repeat(16385 - os.EOL.length - 1)
}));
infos.forEach(info => logger.log(info));
```

Reply:
So synchronous logging calls could overload the stream buffer still without something preventing it then.